From d238d29ce8d91707f99f5d6eaee9dcbc8b08063e Mon Sep 17 00:00:00 2001
From: Salah Triki <salah.triki@gmail.com>
Date: Sat, 19 Jul 2025 16:33:26 +0100
Subject: [PATCH] comedi: Use managed APIs

Replace pci_enable_device() and pci_request_regions() with their
corresponding managed versions to simplify the code.

Signed-off-by: Salah Triki <salah.triki@gmail.com>
---
 drivers/comedi/comedi_pci.c | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/drivers/comedi/comedi_pci.c b/drivers/comedi/comedi_pci.c
index cc2581902195..b8215320e1aa 100644
--- a/drivers/comedi/comedi_pci.c
+++ b/drivers/comedi/comedi_pci.c
@@ -53,14 +53,12 @@ int comedi_pci_enable(struct comedi_device *dev)
 	if (!pcidev)
 		return -ENODEV;
 
-	rc = pci_enable_device(pcidev);
+	rc = pcim_enable_device(pcidev);
 	if (rc < 0)
 		return rc;
 
-	rc = pci_request_regions(pcidev, dev->board_name);
-	if (rc < 0)
-		pci_disable_device(pcidev);
-	else
+	rc = pcim_request_all_regions(pcidev, dev->board_name);
+	if (!rc)
 		dev->ioenabled = true;
 
 	return rc;
@@ -78,12 +76,6 @@ EXPORT_SYMBOL_GPL(comedi_pci_enable);
  */
 void comedi_pci_disable(struct comedi_device *dev)
 {
-	struct pci_dev *pcidev = comedi_to_pci_dev(dev);
-
-	if (pcidev && dev->ioenabled) {
-		pci_release_regions(pcidev);
-		pci_disable_device(pcidev);
-	}
 	dev->ioenabled = false;
 }
 EXPORT_SYMBOL_GPL(comedi_pci_disable);
-- 
2.43.0

